main.go

```diff
@@ -127,22 +129,40 @@ func makeBuckets(m *dto.Metric) map[string]string {
 	return result
 }
 
-func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
+func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily, certificate string, key string) {
 	defer close(ch)
+	var transport *http.Transport
+	if certificate != "" && key != "" {
+		cert, err := tls.LoadX509KeyPair(certificate, key)
+		if err != nil {
+			log.Fatal(err)
+		}
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{cert},
+		}
+		tlsConfig.BuildNameToCertificate()
+		transport = &http.Transport{TLSClientConfig: tlsConfig}
+	} else {
+		transport = &http.Transport{}
+	}
```

Comment:
Things would work without this whole block as a `nil` `Transport` will result in the zero value of `Client`, which is just fine. But we can leave it here for clarity when reading the code.